src/common/xattr_win32.c

```diff
@@ -1,40 +1,194 @@
+#include <errno.h>
+
 #include "common/xattr.h"
+#include "include/compat.h"
 
-#include <errno.h>
+// The Windows headers allocate MAX_PATH + 36 characters
+#define STREAM_MAX_PATH = MAX_PATH + 36;
+
+HANDLE open_stream(const char *path, const char *name, int desired_access,
+                   int create_disposition) {
+  char stream_path[MAX_PATH];
+  snprintf(stream_path, "%s:%s", MAX_PATH, path, name);
+
+  HANDLE handle = CreateFileA(
+    stream_path,
+    desired_access,
+    FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+    NULL,
+    create_disposition,
+    FILE_ATTRIBUTE_NORMAL,
+    NULL);
+  return handle;
+}
+
+BOOL file_exists(const char* path) {
+  DWORD attributes = GetFileAttributesA(path);
+
+  return (attributes != INVALID_FILE_ATTRIBUTES &&
+         !(attributes & FILE_ATTRIBUTE_DIRECTORY));
+}
+
+DWORD get_path_by_fd(int fd, char* path, int size) {
+  HANDLE handle = (HANDLE*)_get_osfhandle(fd);
+  if (handle == INVALID_HANDLE_VALUE) {
+    errno = EINVAL;
+    return -1;
+  }
+
+  if (!GetFinalPathNameByHandleA(handle, path, size,
+                                 FILE_NAME_NORMALIZED)) {
+    return -1;
+  }
+
+  return 0;
+}
 
 int ceph_os_setxattr(const char *path, const char *name,
```

Comment:
IIUC, the ceph_os_*xattr() functions are only used by Filestore, so a port of these for Windows clients shouldn't be necessary.
xattr<->filestreams mapping would likely be a nice way to present CephFS vxattrs in future.

Reply:
Not necessary for the Windows clients, but nice to have if we're ever going to consider running OSDs on Windows. It was a really simple thing to do so I just went on with it.

Anyway, since you moved this out of the common lib, I'm dropping it as well for now, keeping this PR simple.